packages/web3-eth-accounts/src/account.ts

```diff
@@ -1,22 +1,175 @@
 import { utils, getPublicKey } from 'ethereum-cryptography/secp256k1';
+import { keccak256 } from 'ethereum-cryptography/keccak';
+import {
+	TransactionFactory,
+	FeeMarketEIP1559TxData,
+	AccessListEIP2930TxData,
+	TxData,
+} from '@ethereumjs/tx';
+import { ecdsaSign, ecdsaRecover } from 'secp256k1';
 import {
 	toChecksumAddress,
 	bytesToHex,
 	sha3Raw,
 	HexString,
 	isBuffer,
 	isValidString,
+	Address,
+	isHexStrict,
+	utf8ToHex,
+	hexToBytes,
+	isHexString32Bytes,
 } from 'web3-utils';
-import { InvalidPrivateKeyError, PrivateKeyLengthError } from 'web3-common';
+import {
+	InvalidPrivateKeyError,
+	PrivateKeyLengthError,
+	UndefinedRawTransactionError,
+	SignerError,
+	InvalidSignatureError,
+} from 'web3-common';
+import {
+	signatureObject,
+	signFunction,
+	signResult,
+	signTransactionFunction,
+	signTransactionResult,
+} from './types';
 
 // TODO Will be added later
 export const encrypt = (): boolean => true;
 
-// TODO Will be added later
-export const sign = (): boolean => true;
+/**
+ * Hashes the given message. The data will be UTF-8 HEX decoded and enveloped as follows: "\x19Ethereum Signed Message:\n" + message.length + message and hashed using keccak256.
+ */
 
-// TODO Will be added later
-export const signTransaction = (): boolean => true;
+export const hashMessage = (message: string): string => {
+	const messageHex = isHexStrict(message) ? message : utf8ToHex(message);
+
+	const messageBytes = hexToBytes(messageHex);
+
+	const preamble = `\x19Ethereum Signed Message:\n${messageBytes.length}`;
+
+	const ethMessage = Buffer.concat([Buffer.from(preamble), Buffer.from(messageBytes)]);
+
+	return `0x${Buffer.from(keccak256(ethMessage)).toString('hex')}`;
+};
+
+/**
+ * Signs arbitrary data. The value passed as the data parameter will be UTF-8 HEX decoded and wrapped as follows: "\x19Ethereum Signed Message:\n" + message.length + message
+ */
+export const sign = (data: string, privateKey: string): signResult => {
+	const privateKeyParam = privateKey.startsWith('0x') ? privateKey.substring(2) : privateKey;
+
+	if (!isHexString32Bytes(privateKeyParam, false)) {
+		throw new PrivateKeyLengthError();
+	}
+
+	const hash = hashMessage(data);
+
+	const signObj = ecdsaSign(
+		Buffer.from(hash.substring(2), 'hex'),
+		Buffer.from(privateKeyParam, 'hex'),
+	);
+
+	const r = Buffer.from(signObj.signature.slice(0, 32));
+	const s = Buffer.from(signObj.signature.slice(32, 64));
+	const v = signObj.recid + 27;
+
+	return {
+		message: data,
+		messageHash: hash,
+		v: `0x${v.toString(16)}`,
+		r: `0x${r.toString('hex')}`,
+		s: `0x${s.toString('hex')}`,
+		signature: `0x${Buffer.from(signObj.signature).toString('hex')}${v.toString(16)}`,
+	};
+};
+
+/**
+ *  Signs an Ethereum transaction with a given private key.
+ */
+export const signTransaction = (
+	transaction: TxData | AccessListEIP2930TxData | FeeMarketEIP1559TxData,
+	privateKey: string,
+): signTransactionResult => {
```

Comment:
```suggestion
) => {
```

Don't need to specify return type, it's inferred